MarkEditMac/Sources/Editor/Models/EditorDocument.swift

```diff
@@ -225,12 +225,44 @@ extension EditorDocument {
   }
 
   override func writableTypes(for saveOperation: NSDocument.SaveOperationType) -> [String] {
```

Comment:
Based on what I tested: default new file extension, and changing file extension in the save dialog, this seems work as expected.